commands/dkcheck.ml

```diff
@@ -1,141 +1,181 @@
-        ( "--type-lhs",
-          Arg.Set Typing.fail_on_unsatisfiable_constraints,
-          " Forbids rules with untypable left-hand side" );
-        ( "--sr-check",
-          Arg.Int (fun i -> Srcheck.srfuel := i),
-          "LVL Sets the level of subject reduction checking to LVL.\n\
-          \                   Default value is 1. Values < 0 may not terminate \
-           on\n\
-          \                   rules that do not preserve typing. " );
-        ( "--snf",
-          Arg.Set Env.errors_in_snf,
-          " Normalizes all terms printed in error messages" );
-        ( "--db",
-          Arg.Set Pp.print_db_enabled,
-          " Prints De Bruijn indices in error messages" );
-        ( "--confluence",
-          Arg.String Confluence.set_cmd,
-          "CMD Set the external confluence checker command to CMD" );
-        ( "--beautify",
-          Arg.Set beautify,
-          " Pretty printer. Print on the standard output" );
-        ( "--version",
-          Arg.Unit (fun () -> Format.printf "Dedukti %s@." Version.version),
-          " Prints the version number" )
-        (* Deprecated flags. TODO: Remove them from the argument parsing. *);
-        deprecated "-errors-in-snf" "--snf" (Arg.Set Env.errors_in_snf);
-        deprecated "-cc" "--confluence" (Arg.String Confluence.set_cmd);
-        deprecated "-eta" "--eta"
-          (Arg.Tuple [Arg.Set Reduction.eta; Arg.Clear Env.check_arity]);
-        deprecated "-coc" "--coc" (Arg.Set Typing.coc);
-        deprecated "-nl" "no flag" (Arg.Unit ignore);
-        deprecated "-version" "--version"
-          (Arg.Unit (fun () -> Format.printf "Dedukti %s@." Version.version));
-      ]
+  Arg.(value & opt string "" & info ["d"; "debug"] ~docv:"FLAGS" ~doc ~docs)
+
+let quiet =
+  let doc = "Do not print anything (same as -d q)" in
+  let docs = Manpage.s_common_options in
+  Arg.(value & flag & info ["q"; "quiet"] ~doc ~docs)
+
+let verbose =
+  let doc = "Verbose mode (equivalent to -d montru). Overrides -d/--debug." in
+  let docs = Manpage.s_common_options in
+  Arg.(value & flag & info ["v"; "verbose"] ~doc ~docs)
+
+let config_t =
+  let fn no_color debug quiet verbose = {no_color; debug; quiet; verbose} in
+  Term.(const fn $ no_color $ debug $ quiet $ verbose)
+
+let init (c : config) : unit =
+  (* Verbose overrides debug that overrides quiet. *)
+  if c.quiet then Env.set_debug_mode "q";
+  Env.set_debug_mode c.debug;
+  if c.verbose then Env.set_debug_mode "montru";
+  Errors.color := not c.no_color
+
+(** dkcheck options *)
+
+let incl =
+  let doc = "Add directory $(docv) to the load path" in
+  Arg.(value & opt_all dir [] & info ["I"; "include"] ~docv:"DIR" ~doc)
+
+let export =
+  let doc = "Generate object files (\".dko\")" in
+  Arg.(value & flag & info ["e"] ~doc)
+
+let beautify =
+  (* TODO put in a separate command. *)
+  let doc = "Pretty print a file on the standard input" in
+  Arg.(value & flag & info ["beautify"] ~doc)
+
+let run_on_stdin =
```

Comment:
I think you should document what happen if `run_on_stdin` is set while we give at the same time a file on the command-line.